fineract-provider/src/main/java/org/apache/fineract/cob/loan/StayedLockedLoansTasklet.java

```diff
@@ -35,14 +40,24 @@ public class StayedLockedLoansTasklet implements Tasklet {
 
     private final LoanAccountLockRepository loanAccountLockRepository;
     private final BusinessEventNotifierService businessEventNotifierService;
+    private final LoanRepository loanRepository;
 
     @Override
     public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
         List<LoanAccountLock> loanAccountLocks = loanAccountLockRepository.findAll();
         if (!loanAccountLocks.isEmpty()) {
             List<Long> loanIds = loanAccountLocks.stream().map(LoanAccountLock::getLoanId).toList();
-            businessEventNotifierService.notifyPostBusinessEvent(new LoanAccountsStayedLockedBusinessEvent(loanIds));
+            businessEventNotifierService.notifyPostBusinessEvent(new LoanAccountsStayedLockedBusinessEvent(buildLoanAccountData(loanIds)));
         }
         return RepeatStatus.FINISHED;
     }
+
+    private LoanAccountsStayedLockedBusinessEvent.Data buildLoanAccountData(List<Long> loanIds) {
+        List<LoanIdAndExternalIdAndAccountNo> stayedLockedLoanAccounts = loanRepository.findAllStayedLockedByLoanIds(loanIds);
```

Comment:
Please make sure the loanIds array is not too big / request. Postgres has some restrictions on the parameter size.

Reply:
Do we need to at all go to the database to grab this data? Can't we pass this through some context?

Reply:
I changed the query, we don't have to deal with the limitation anymore.